src/order/antisymmetrization.lean

```diff
@@ -0,0 +1,175 @@
+  decidable_eq := classical.dec_rel _,
+  decidable_le := classical.dec_rel _,
+  decidable_lt := classical.dec_rel _,
+  ..antisymmetrization.partial_order }
+
+@[simp] lemma to_antisymmetrization_le_to_antisymmetrization_iff :
+  to_antisymmetrization a ≤ to_antisymmetrization b ↔ a ≤ b := iff.rfl
+
+@[simp] lemma to_antisymmetrization_lt_to_antisymmetrization_iff :
+  to_antisymmetrization a < to_antisymmetrization b ↔ a < b := iff.rfl
+
+@[simp] lemma of_antisymmetrization_le_of_antisymmetrization_iff {a b : antisymmetrization α} :
+  of_antisymmetrization a ≤ of_antisymmetrization b ↔ a ≤ b :=
+by convert to_antisymmetrization_le_to_antisymmetrization_iff.symm;
+  exact (to_antisymmetrization_of_antisymmetrization _).symm
+
+@[simp] lemma of_antisymmetrization_lt_of_antisymmetrization_iff {a b : antisymmetrization α} :
+  of_antisymmetrization a < of_antisymmetrization b ↔ a < b :=
+by convert to_antisymmetrization_lt_to_antisymmetrization_iff.symm;
+  exact (to_antisymmetrization_of_antisymmetrization _).symm
+
+lemma to_antisymmetrization_mono : monotone (@to_antisymmetrization α _) := λ a b, id
+
+/-- `to_antisymmetrization` as an order homomorphism. -/
+@[simps] def order_hom.to_antisymmetrization : α →o antisymmetrization α :=
+⟨to_antisymmetrization, λ a b, id⟩
+
+private lemma lift_fun_le_equiv (f : α →o β) :
+  ((le_equiv.setoid α).r ⇒ (le_equiv.setoid β).r) f f :=
+λ a b h, ⟨f.mono h.1, f.mono h.2⟩
+
+/-- Turns an order homomorphism from `α` to `β` into one from `antisymmetrization α` to
+`antisymmetrization β`. `antisymmetrization` is actually a functor. See `Preorder_to_PartialOrder`.
+-/
+protected def order_hom.antisymmetrization (f : α →o β) :
+  antisymmetrization α →o antisymmetrization β :=
+⟨quotient.map' f $ lift_fun_le_equiv f, λ a b, quotient.induction_on₂' a b $ f.mono⟩
+
+@[simp] lemma order_hom.coe_antisymmetrization (f : α →o β) :
+  ⇑f.antisymmetrization = quotient.map' f (lift_fun_le_equiv f) := rfl
+
+@[simp] lemma order_hom.antisymmetrization_apply (f : α →o β) (a : antisymmetrization α) :
+  f.antisymmetrization a = quotient.map' f (lift_fun_le_equiv f) a := rfl
+
+@[simp] lemma order_hom.antisymmetrization_apply_mk (f : α →o β) (a : α) :
+  f.antisymmetrization (to_antisymmetrization a) = (to_antisymmetrization (f a)) :=
+quotient.map'_mk' f (lift_fun_le_equiv f) _
+
+variables (α)
+
+/-- `of_antisymmetrization` as an order embedding. -/
+@[simps] noncomputable def order_embedding.of_antisymmetrization : antisymmetrization α ↪o α :=
+{ to_fun := of_antisymmetrization,
+  inj' := λ _ _, quotient.out_inj.1,
+  map_rel_iff' := λ a b, of_antisymmetrization_le_of_antisymmetrization_iff }
+
+/-- `antisymmetrization` and `order_dual` commute. -/
+def order_iso.dual_antisymmetrization :
+  order_dual (antisymmetrization α) ≃o antisymmetrization (order_dual α) :=
+{ to_fun := quotient.map' id $ λ _ _, and.symm,
+  inv_fun := quotient.map' id $ λ _ _, and.symm,
```

Comment:
This really ought to have an invocation of `to_dual` or `of_dual` in it somewhere.

Can you also add the lemmas that say
```
dual_antisymmetrization (to_dual (to_antisymmetrization x)) = to_antisymmetrization (to_dual x)
```
and similarly for `symm`?

Reply:
It's a real pain to do it proper because of the quotient stuff, so I will just write the simp lemmas with the correct form.